src/hooks/transfers/useTransfers.tsx

```diff
@@ -0,0 +1,276 @@
+    queryKey: ['transferEligibleAssets'],
+    queryFn: async () => {
+      const assetsByChain = await skipClient.assets({
+        includeEvmAssets: true,
+        includeSvmAssets: true,
+      });
+      return { assetsByChain };
+    },
+    refetchOnWindowFocus: false,
+    refetchOnMount: false,
+    refetchOnReconnect: false,
+  });
+  const { chainsByNetworkMap = {} } = chainsQuery.data ?? {};
+  const { assetsByChain = {} } = assetsQuery.data ?? {};
+
+  const walletNetworkType = getNetworkTypeFromWalletInfo(sourceAccount?.walletInfo);
+  const selectedChainId = transferType === TransferType.Deposit ? fromChainId : toChainId;
+
+  const toToken = useMemo(() => {
+    return assetsByChain[toChainId ?? '']?.find((token) => token.denom === toTokenDenom);
+  }, [toChainId, toTokenDenom, assetsByChain]);
+
+  const fromToken = useMemo(() => {
+    return assetsByChain[fromChainId ?? '']?.find((token) => token.denom === fromTokenDenom);
+  }, [fromChainId, fromTokenDenom, assetsByChain]);
+
+  const chainsForNetwork = useMemo(
+    () => chainsByNetworkMap?.[walletNetworkType] ?? [],
+    [walletNetworkType, chainsByNetworkMap]
+  );
+
+  const assetsForSelectedChain = useMemo(() => {
+    if (selectedChainId) {
+      // sort native denoms to the front of list
+      const sortedAssetsForChain = (assetsByChain[selectedChainId] ?? []).sort((a) => {
+        return isNativeDenom(a.denom) ? -1 : 1;
+      });
+      return sortedAssetsForChain;
+    }
+    return [];
+  }, [selectedChainId, assetsByChain]);
+
+  const defaultChainId = useMemo(() => {
+    return getDefaultChainIDFromNetworkType(walletNetworkType) ?? chainsForNetwork[0]?.chainID;
+  }, [walletNetworkType, chainsForNetwork]);
+
+  const defaultTokenDenom = useMemo(() => {
+    return getDefaultTokenDenomFromAssets(assetsForSelectedChain);
+  }, [assetsForSelectedChain]);
+
+  const hasAllParams =
+    !!fromToken?.denom &&
+    !!toToken?.denom &&
+    !!fromToken?.decimals &&
+    !!toToken?.decimals &&
+    !!fromChainId &&
+    !!toChainId &&
+    !!fromAddress &&
+    !!toAddress &&
+    !!transferType &&
+    !!amount &&
+    !!dydxAddress;
+
+  const routeQuery = useQuery({
+    queryKey: [
+      'transferRoute',
+      fromToken,
+      toToken,
+      fromChainId,
+      toChainId,
+      fromAddress,
+      toAddress,
+      transferType,
+      amount,
+      dydxAddress,
+      selectedDydxChainId,
+    ],
+    queryFn: async () => {
+      // this should never happen, this is just to satisfy typescript
+      // react queries should never return null.
+      if (!hasAllParams) return null;
+
+      const baseParams = {
+        sourceAssetDenom: fromToken.denom,
+        sourceAssetChainID: fromToken.chainID,
+        destAssetDenom: toToken.denom,
+        destAssetChainID: toToken.chainID,
+        allowUnsafe: true,
+        slippageTolerancePercent: '1',
+        smartRelay: true,
+        amountIn: parseUnits(amount, fromToken.decimals ?? 0).toString(),
```

Comment:
i'll add a comment to talk to skip about this, i have a few things on my list rn including this. my suspicion is that it's just mistyped

Reply:
i think the more appropriate thing to do here is to either throw an error or not let the user proceed with a swap if we don't have token decimal information, else you run into the risk of trading an unknown number of the user's tokens on accident

Reply:
i like that - until i get an answer from skip i'll gate this with an error 👍

Reply:
ah i remember now, we do actualyl check for this in hasAllParams, but we just need a specific check since typescript can't be certain the property hasn't been updated since `hasAllParams` was declared. i'll just add an explicit check next to `!hasAllParams)